registry/spec.md

```diff
@@ -1491,71 +1632,1350 @@ An attempt to delete all versions MUST generate an error.
-extension attributes:
-
+``` JSONC
+{
+  "$schema": "https://cloudevents.io/schemas/registry",
+  "specversion": "0.4-wip",
+  "id": "urn:uuid:3978344f-8596-4c3a-a978-8fc9a6a469f7",
+  "endpoints" : 
+  {
+    "com.example.telemetry" : {
+      "type": "endpoint",
+      "id" : "com.example.telemetry",
+      "usage": "consumer",
+      "config": {
+        // ... details ...
+      },
+      "format" : "CloudEvents/1.0"
+      "definitiongroups" :[
+          "https://site.example.com/registry/definitiongroups/com.example.telemetryEvents"
+      ]
+    }
+  }
+}
 ```
-"origin": "URI", ?
-"deprecated": { ... }, ?
-"channel", "STRING", ?
 
-"usage": "subscriber|consumer|producer",
-"config": {
-  "protocol": "STRING",
-  "endpoints": "URL" | [ "URL", ... ],
-  "options": { ... }, ?
-  "strict": true|false ?
-}, ?
+If the message definitions and schemas are stored in a file-based registry,
+including files shared via public version control repositories, the reference
+link will first reference the file and then the object within the file, using
+[JSON Pointer][JSON Pointer] syntax:
 
-"format": "STRING", ?
-"definitionGroups": [ DEFINITIONGROUP-URI-Reference, ... ] ?
-```
+``` JSONC
+{
+  "$schema": "https://cloudevents.io/schemas/registry",
+  "specversion": "0.4-wip",
+  "id": "urn:uuid:3978344f-8596-4c3a-a978-8fc9a6a469f7",
+  "endpoints" : 
+  {
+    "com.example.telemetry" : {
+      "type": "endpoint",
+      "id" : "com.example.telemetry",
+      "usage": "consumer",
+      "config": {
+        // ... details ......
+      },
+      "format" : "CloudEvents/1.0"
+      "definitiongroups" :[
+        "https://rawdata.repos.example.com/myorg/myproject/main/example.telemetryEvents.cereg#/definitiongroups/com.example.telemetryEvents"
+      ]
+    }
+  }
+}
+```
+
+All other references to other objects in the registry can be expressed in the
+same way.
+
+While the CloudEvents Registry is primarily motivated by enabling development of
+CloudEvents-based event flows, the registry is not limited to CloudEvents. It
+can be used to describe any asynchronous messaging or eventing endpoint and its
+messages, including such that do not use CloudEvents at all. The [Message
+Formats](#message-formats) section therefore not only describes the attribute
+meta-schema for CloudEvents, but also meta-schemas for the native message
+envelopes of MQTT, AMQP, and other messaging protocols.
+
+The registry is designed to be extensible to support any structured data
+encoding and related schemas for message or event payloads. The [Schema
+Formats](#schema-formats) section describes the meta-schema for JSON Schema, XML
+Schema, Apache Avro schema, Protobuf schema.
+
+### File format
+
+A CloudEvents Registry can be implemented using the Registry API or with plain
+text files.
+
+When using the file-based model, files with the extension `.cereg` use JSON
+encoding. Files with the extension `.cereg.yaml` or `.cereg.yml` use YAML
```

Comment:
.cereg.yaml is unfortunate, but I understand why you did it.
Not sure I prefer this idea but just to brainstorm... is the double extension idea better than:
- foo-cereg.json / foo-cereg.yaml
- cereg-foo.json / cereg-foo.yaml

Or some variant like that where the "reg" part is included in the base name rather than the extension.